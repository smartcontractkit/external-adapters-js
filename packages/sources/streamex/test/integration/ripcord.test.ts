import {
  setEnvVariables,
  TestAdapter,
} from '@chainlink/external-adapter-framework/util/testing-utils'
import * as nock from 'nock'
import { mockReserveResponseRipcordFailure } from './fixtures'

describe('execute', () => {
  let spy: jest.SpyInstance
  let testAdapter: TestAdapter
  let oldEnv: NodeJS.ProcessEnv

  beforeAll(async () => {
    oldEnv = JSON.parse(JSON.stringify(process.env))
    const mockDate = new Date('2001-01-01T11:11:11.111Z')
    spy = jest.spyOn(Date, 'now').mockReturnValue(mockDate.getTime())

    process.env.API_KEY = 'streamex-api-key'

    const adapter = (await import('../../src')).adapter
    adapter.rateLimiting = undefined
    testAdapter = await TestAdapter.startWithMockedCache(adapter, {
      testAdapter: {} as TestAdapter<never>,
    })
  })

  afterAll(async () => {
    setEnvVariables(oldEnv)
    await testAdapter.api.close()
    nock.restore()
    nock.cleanAll()
    spy.mockRestore()
  })

  describe('reserve endpoint when ripcord true', () => {
    it('should fail with 502 error', async () => {
      const data = {
        endpoint: 'reserve',
      }
      mockReserveResponseRipcordFailure()
      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(502)
      expect(response.json()).toMatchSnapshot()
    })
  })
})
