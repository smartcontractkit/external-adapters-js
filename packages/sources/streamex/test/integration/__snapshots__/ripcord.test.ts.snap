// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute reserve endpoint when ripcord true should fail with 502 error 1`] = `
{
  "errorMessage": "Ripcord indicator true. Details: Balances",
  "ripcord": true,
  "ripcordAsInt": 1,
  "ripcordDetails": "Balances",
  "statusCode": 502,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
    "providerIndicatedTimeUnixMs": 1760583701362,
  },
}
`;
